Analyser/src/bsa_analyser/github/io/FileValidator.java

```diff
@@ -5,24 +5,79 @@
  */
 package bsa_analyser.github.io;
 
-import java.io.BufferedReader;
-import java.io.FileReader;
+import java.io.*;
+
 import static java.lang.Boolean.FALSE;
 import static java.lang.Boolean.TRUE;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Scanner;
 import javax.swing.JOptionPane;
 
 /**
  *
- * @author jkell
+ * @author jkelly, Valetina, and Tolu
  */
 
 
 public class FileValidator {
-    
-    
+    public static void fasta_content_checker(File user_file) {
+
+    boolean b = false;
+
+    List<String> Fasta_Content = new ArrayList<String>();
+        if (user_file.getName().endsWith("fasta") || user_file.getName().endsWith("fa")) {
+            String line = "";
+            try {
+
+                BufferedReader br = new BufferedReader(new FileReader((user_file)));
+                while (((line = br.readLine()) != null)) {
+
+                    if ((!line.contains("A|T|G|C|N")) && (!line.startsWith(">"))) {
```

Comment:
This line here is checking if the file does not contain the string "A|T|G|C|N" not the individual bases. So this is always going to be True unless the Fasta file actually contains that string. Also a fasta file has two types of lines either it starts with '>' or has the nucleotides in the sting. This logic is checking for both.